### PR TITLE
Misc/upgrade nelmio doc

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

**Comment by praneethpr** [comment]
I did a good end to end test for this branch, this could be merged to Symfony 4.4 upgrade. If we do it at another time, we should do an end-to end test again because these changes affect the APIs @mickel1138 @fightingtexasaggie

File: composer.json [hunk 1]
```diff
@@ -41,11 +42,12 @@
         "symfony/security-bundle": "4.4.*",
         "symfony/swiftmailer-bundle": "^3.1",
         "symfony/templating": "4.4.*",
+        "symfony/twig-pack": "^1.0",
         "symfony/validator": "4.4.*",
         "symfony/webpack-encore-bundle": "^1.1",
         "symfony/yaml": "4.4.*",
         "twig/extensions": "^1.5",
-        "willdurand/hateoas-bundle": "^1.1"
+        "willdurand/hateoas-bundle": "^2.1"
```
Comment (mickel1138) [inline composer.json]: Maybe make this 2.0, so we know that we specifically needed 2.1 (or higher)

File: config/packages/nelmio_api_doc.yaml [hunk 2]
```diff
@@ -1,22 +1,9 @@
...
-#            license:              null
-#            licenseUrl:           null
-#    cache:
-#        enabled:              false
-#        file:                 '%kernel.cache_dir%/api-doc.cache'
+nelmio_api_doc:
+    documentation:
+        info:
+            title: Pelagos
+            description: This is an awesome app!
```
Comment (mickel1138) [inline config/packages/nelmio_api_doc.yaml]: Not sure what the description should be, or the version?

File: config/routes/nelmio_api_doc.yaml [hunk 3]
```diff
@@ -1,3 +1,12 @@
...
-    resource: "@NelmioApiDocBundle/Resources/config/routing.yml"
-    prefix:   /api/doc
+# Expose your documentation as JSON swagger compliant
+app.swagger:
+    path: /api/doc.json
+    methods: GET
+    defaults: { _controller: nelmio_api_doc.controller.swagger }
+
+## Requires the Asset component and the Twig bundle
+## $ composer require twig asset
```
Comment (mickel1138) [inline config/routes/nelmio_api_doc.yaml]: Should we require this (twig) separately? It's already installed because of other packages.

Comment (mickel1138) [reply config/routes/nelmio_api_doc.yaml]: Actually, I see it has been added.

File: composer.json (hunk 1 above)
Comment (praneethpr) [reply composer.json]: ^2.1 does mean that we specifically needed 2.1 or higher right?

File: config/packages/nelmio_api_doc.yaml (hunk 2 above)
Comment (praneethpr) [reply config/packages/nelmio_api_doc.yaml]: we can start with 1.0.0 for version and its anyways for internal viewing

File: config/routes/nelmio_api_doc.yaml (hunk 3 above)
Comment (praneethpr) [reply config/routes/nelmio_api_doc.yaml]: This installs a different package called symfony/twig-pack, I think its different from the twig package we use?

File: composer.json (hunk 1 above)
Comment (mickel1138) [reply composer.json]: Yes, it does mean that. But this makes me thing that 2.0 was no good. What is there is no 2.1? It was more of a type-o I think.

File: config/packages/nelmio_api_doc.yaml (hunk 2 above)
Comment (mickel1138) [reply config/packages/nelmio_api_doc.yaml]: The API is public, some people may make use of it. However, we don't publish it the URL publicly I thnk.

File: config/routes/nelmio_api_doc.yaml (hunk 3 above)
Comment (mickel1138) [reply config/routes/nelmio_api_doc.yaml]: The symfony pack is just a bundle/package of commonly used bundles/packages that Symfony uses when using twig.

**Pull request merged**